core/error_macros.h

```diff
@@ -115,27 +115,40 @@ extern bool _err_error_exists;
 #endif
 
 #define ERR_FAIL_INDEX(m_index, m_size)                                                                                    \
-	do {                                                                                                                   \
+	{                                                                                                                      \
```

Comment:
Is anyone getting a warning about that? If not, I think surprising constructs like this would be better dropped.